### DISCUSSION TITLE
Using remote.it instead of dynamic DNS (with Wireguard)

### DISCUSSION THREAD

**Discussion by foice** [pivpn/pivpn#1444] [Support]
I would like to avoid a dynamic DNS service and use instead a service that opens a proxy connection to my VPN server _on demand_. I use this type of strategy for SSH log in to my machines with [remote.it](https://app.remote.it/) and it works fine for that.

I can setup a remote.it connection that, upon request, will generate an address of the type `proxyNN.rt3.io:XYZTS` that corresponds to a UDP port on my VPN client (the standard 51820 in my case). This is _almost_  a replacement for a dynamic DNS, as I can reach a public address that tunnels to the port where Wireguard is listening. The reason why it is not the same as dynamic DNS is that the public address changes all the time, i.e. I get different `NN` and `port` every time I trigger a new remote.it connection. 

My question is how to deal with this changing, but knowable IP and Port. 

My understanding is that I need to
- make a .conf file *once*
-  add it to my device (android Wireguard client)
-  trigger my remote.it tunnel to UDP 51820
- change the server address on the Android Wireguard client
- Turn on the Wireguard connection.

Did I miss any step? For instance it is not clear to me if I need to change `/etc/pivpn/wireguard/setupVars.conf` everytime I get a new `proxyNN.rt3.io:XYZTS` address.

I am interested also to OpenVPN if it can work in this setup, but I doubt as I have made successful connection using OpenVPN but I had to open [TCP 443, TCP 943, TCP 945 and UDP 1194](https://openvpn.net/vpn-server-resources/install-openvpn-access-server-on-raspberry-pi/). Opening so many ports at once with remote.it it's going to be very long and impractical. So I choose Wireguard for it seems to need *only* 51820, maybe I was wrong ... ?

**Comment by orazioedoardo** [comment]
> This is almost a replacement for a dynamic DNS, as I can reach a public address that tunnels to the port where Wireguard is listening. The reason why it is not the same as dynamic DNS is that the public address changes all the time, i.e. I get different NN and port every time I trigger a new remote.it connection.

I'm not familiar with this service, but I've looked around and it should work. Have you tried?

> Did I miss any step? For instance it is not clear to me if I need to change /etc/pivpn/wireguard/setupVars.conf everytime I get a new proxyNN.rt3.io:XYZTS address.

That file is the "template", it's used to generate new clients, it doesn't affect existing clients.

> I am interested also to OpenVPN if it can work in this setup, but I doubt as I have made successful connection using OpenVPN but I had to open TCP 443, TCP 943, TCP 945 and UDP 1194.

Those ports are required for the access server. The daemon just need one tcp port or one udp port.
